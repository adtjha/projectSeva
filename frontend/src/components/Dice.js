import { Transition } from '@headlessui/react'
import { Fragment } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import one from '../images/dice/1.svg'
import two from '../images/dice/2.svg'
import three from '../images/dice/3.svg'
import four from '../images/dice/4.svg'
import five from '../images/dice/5.svg'
import six from '../images/dice/6.svg'
import {
    getShowing,
    rolled,
    set_dice,
    set_rolled,
    set_showing,
} from '../store/dice'

const Dice = (props) => {
    let isShowing = useSelector(getShowing)
    let hasRolled = useSelector(rolled)

    const srcList = [one, two, three, four, five, six]

    const dispatch = useDispatch()

    const handleClick = () => {
        if (!hasRolled) {
            dispatch(set_showing(false))
            setTimeout(() => {
                dispatch(set_dice())
                dispatch(set_rolled(true))
                dispatch(set_showing(true))
            }, 500)
        } else {
            console.log('PLAY MOVE, DICE ROLLED ONCE', hasRolled)
        }
    }

    return (
        <div className="flex flex-col items-center py-8">
            <div className="w-32 h-32">
                <Transition
                    as={Fragment}
                    show={isShowing}
                    enter="transform transition duration-300"
                    enterFrom="opacity-0 -rotate-180 scale-50"
                    enterTo="opacity-100 rotate-0 scale-100"
                    leave="transform duration-200 transition ease-in-out"
                    leaveFrom="opacity-100 rotate-0 scale-100 "
                    leaveTo="opacity-0 scale-95 "
                >
                    <div
                        className="w-16 font-mono text-4xl m-auto text-center p-0 text-gray-400"
                        onClick={handleClick}
                    >
                        <img src={srcList[props.num - 1]} alt={props.num}></img>
                    </div>
                </Transition>
            </div>
        </div>
    )
}

export default Dice
