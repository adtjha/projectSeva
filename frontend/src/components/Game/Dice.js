import { Transition } from '@headlessui/react'
import { Fragment } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import one from '../../images/dice/1.svg'
import two from '../../images/dice/2.svg'
import three from '../../images/dice/3.svg'
import four from '../../images/dice/4.svg'
import five from '../../images/dice/5.svg'
import six from '../../images/dice/6.svg'
import { fetch_dice, getShowing, rolled } from '../../store/dice'
import { getColor, getGameCurrentPlayer, getGameId } from '../../store/user'

const Dice = (props) => {
    let isShowing = useSelector(getShowing)
    let hasRolled = useSelector(rolled)

    const gameId = useSelector(getGameId)
    const userColor = useSelector(getColor)
    const currentColor = useSelector(getGameCurrentPlayer)
    const isChance = userColor === currentColor

    const srcList = [one, two, three, four, five, six]

    const dispatch = useDispatch()

    const handleClick = () => {
        if (!hasRolled && isChance) {
            dispatch(fetch_dice({ gameId, userColor }))
        } else if (isChance) {
            console.log(
                'PLAY MOVE, DICE ROLLED ONCE',
                { hasRolled },
                { isChance }
            )
        } else {
            console.log('Not Your Chance', { isChance })
        }
    }

    // useLogger(
    //     'Dice',
    //     rendersCount,
    //     { isShowing },
    //     { hasRolled },
    //     { isChance },
    //     props.num
    // )

    return (
        <div className="w-16 h-16 mx-auto my-8 lg:my-auto">
            <Transition
                as={Fragment}
                show={isShowing}
                enter="transform transition duration-500"
                enterFrom="opacity-0 -rotate-360 scale-50"
                enterTo="opacity-100 rotate-0 scale-100"
                leave="transform duration-500 transition ease-in-out"
                leaveFrom="opacity-100 rotate-0 scale-100 "
                leaveTo="opacity-0 scale-95 "
            >
                <div
                    className="w-16 h-16 font-mono text-4xl m-auto text-center p-0 text-gray-400"
                    onClick={handleClick}
                >
                    <img src={srcList[props.num - 1]} alt={props.num}></img>
                </div>
            </Transition>
         </div>
    )
}

export default Dice
