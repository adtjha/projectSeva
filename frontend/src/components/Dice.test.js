import { render } from '@testing-library/react'
import React, { Fragment } from 'react'
import reactDom from 'react-dom'
import Dice from './Dice'
import one from '../images/dice/1.svg'
import two from '../images/dice/2.svg'
import three from '../images/dice/3.svg'
import four from '../images/dice/4.svg'
import five from '../images/dice/5.svg'
import six from '../images/dice/6.svg'
import { Transition } from '@headlessui/react'


// TEST =================================================================================================
test('<Dice /> matches snapshot', () => {
    const num = 1
    const srcList = [one, two, three, four, five, six]
    
    const component = render(
        <div className="flex flex-col items-center py-8">
            <div className="w-32 h-32">
                <Transition
                    as={Fragment}
                    show={true}
                    enter="transform transition duration-300"
                    enterFrom="opacity-0 -rotate-180 scale-50"
                    enterTo="opacity-100 rotate-0 scale-100"
                    leave="transform duration-200 transition ease-in-out"
                    leaveFrom="opacity-100 rotate-0 scale-100 "
                    leaveTo="opacity-0 scale-95 "
                >
                    <div
                        className="w-16 font-mono text-4xl m-auto text-center p-0 text-gray-400"
                        onClick={() => {}}
                    >
                        <img src={srcList[num-1]} alt={num}></img>
                    </div>
                </Transition>
            </div>
        </div>
    )

    expect(component.container).toMatchSnapshot()
})

// TEST =================================================================================================
test('<Dice /> is rendering without crashing', () => {
    const div = document.createElement('div')
    render(
        <div className="flex flex-col items-center py-8">
            <div className="w-32 h-32">
                <Transition
                    as={Fragment}
                    show={true}
                    enter="transform transition duration-300"
                    enterFrom="opacity-0 -rotate-180 scale-50"
                    enterTo="opacity-100 rotate-0 scale-100"
                    leave="transform duration-200 transition ease-in-out"
                    leaveFrom="opacity-100 rotate-0 scale-100 "
                    leaveTo="opacity-0 scale-95 "
                >
                    <div
                        className="w-16 font-mono text-4xl m-auto text-center p-0 text-gray-400"
                        onClick={() => {}}
                    >
                        <img src={one} alt={1}></img>
                    </div>
                </Transition>
            </div>
        </div>,
        div
    )
})

// TEST =================================================================================================
test('Defaults to one', () => {
    const div = document.createElement('div')
    render(
        <div className="flex flex-col items-center py-8">
            <div className="w-32 h-32">
                <Transition
                    as={Fragment}
                    show={true}
                    enter="transform transition duration-300"
                    enterFrom="opacity-0 -rotate-180 scale-50"
                    enterTo="opacity-100 rotate-0 scale-100"
                    leave="transform duration-200 transition ease-in-out"
                    leaveFrom="opacity-100 rotate-0 scale-100 "
                    leaveTo="opacity-0 scale-95 "
                >
                    <div
                        className="w-16 font-mono text-4xl m-auto text-center p-0 text-gray-400"
                        onClick={() => {}}
                    >
                        <img src={one} alt={1}></img>
                    </div>
                </Transition>
            </div>
        </div>,
        div
    )
})

// TEST =================================================================================================
test('roll dice on dice_roll event', () => {})

// TEST =================================================================================================
test('click to roll dice if not hasRolled', () => {})
