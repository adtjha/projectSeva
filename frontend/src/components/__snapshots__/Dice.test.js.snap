// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dice /> matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center py-8"
  >
    <div
      class="w-32 h-32"
    >
      <div
        class="w-16 font-mono text-4xl m-auto text-center p-0 text-gray-400 transform transition duration-300 opacity-0 -rotate-180 scale-50"
      >
        <img
          alt="1"
          src="1.svg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Dice matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center py-8"
  >
    <div
      class="w-32 h-32"
    >
      <div
        class="w-16 font-mono text-4xl m-auto text-center p-0 text-gray-400 transform transition duration-300 opacity-0 -rotate-180 scale-50"
      >
        <img
          alt="1"
          src="1.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
